### PR TITLE
Fix Oracle check

### LINKED ISSUE #9457: [L-01] Missing oracle can cause unexpected behavior

### PR DESCRIPTION
**Pull request by martinvol**
### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #9457 

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### Documentation

_The set of community facing docs that have been added/modified because of this change_

### CODE REVIEW

File: packages/protocol/contracts/stability/Reserve.sol [hunk 1]
```diff
@@ -538,9 +538,13 @@ contract Reserve is
       uint256 stableAmount;
       uint256 goldAmount;
       (stableAmount, goldAmount) = sortedOracles.medianRate(_tokens[i]);
-      uint256 stableTokenSupply = IERC20(_tokens[i]).totalSupply();
-      uint256 aStableTokenValueInGold = stableTokenSupply.mul(goldAmount).div(stableAmount);
-      stableTokensValueInGold = stableTokensValueInGold.add(aStableTokenValueInGold);
+
+      if (goldAmount != 0) {
```
Comment (martinvol) [inline packages/protocol/contracts/stability/Reserve.sol:542]: @m-chrzan Do you think I should add a check that the supply of the stabletoken should be zero as well?

File: packages/protocol/test/stability/reserve.ts [hunk 2]
```diff
@@ -781,7 +781,7 @@ contract('Reserve', (accounts: string[]) => {
     })
   })
 
-  describe('#getReserveRatio', () => {
+  describe.only('#getReserveRatio', () => {
```
Comment (martinvol) [inline packages/protocol/test/stability/reserve.ts:784]: FIXME

File: packages/protocol/test/stability/reserve.ts [hunk 3]
```diff
@@ -804,6 +804,26 @@ contract('Reserve', (accounts: string[]) => {
         value: reserveGoldBalance,
       })
     })
+
+    describe.only('works with stabletoken with no report', async () => {
```
Comment (martinvol) [inline packages/protocol/test/stability/reserve.ts:808]: FIXME

**Review by m-chrzan — APPROVED** [review]
Remember to remove the `.only`s.

File: packages/protocol/test/stability/reserve.ts (hunk 2 above)
Comment (martinvol) [inline packages/protocol/test/stability/reserve.ts]:
```suggestion
  describe('#getReserveRatio', () => {
```

File: packages/protocol/test/stability/reserve.ts (hunk 3 above)
Comment (martinvol) [inline packages/protocol/test/stability/reserve.ts]:
```suggestion
    describe('works with stabletoken with no report', async () => {
```

**Pull request merged**